packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -212,6 +245,145 @@ export class TerminalFrontendContribution implements FrontendApplicationContribu
         });
     }
 
+    async onStart(app: FrontendApplication): Promise<void> {
+        await this.contributeDefaultProfiles();
+
+        this.terminalPreferences.onPreferenceChanged(e => {
+            if (e.preferenceName.startsWith('terminal.integrated.')) {
+                this.mergePreferencesPromise = this.mergePreferencesPromise.finally(() => this.mergePreferences());
+            }
+        });
+        this.mergePreferencesPromise = this.mergePreferencesPromise.finally(() => this.mergePreferences());
+
+        this.profileService.onAdded(id => {
+            // extension contributions get read after this point: need to set the default profile if necessary
+            let defaultProfileId;
+            switch (OS.type()) {
```

Comment:
Suggestion: Given that the terminal runs on the backend, it makes more sense to decide on the preference by using the backend OS, instead of the frontend OS.

```suggestion
            switch (OS.backend.type()) {
```

Reply:
That's not a suggestion: it's a bug.